docs/proposals/scale_group_upgrade.md

```diff
@@ -0,0 +1,141 @@
+
+The upgrade process would look like this:
+* Create a new scale group `compute 456`
+* Scale up the new scale group to match the first groups capacity
+* Allow nodes to join the cluster by approving node CSRs
+* Set the nodes from `compute 123` as unschedulable
+* Drain the `compute 123` pods from the nodes
+* Determine if we want to role back to `compute 123` or continue on with `compute 456`.
+* Remove scale group `compute 123` or `compute 456`
+
+## Benefits
+
+There are many benefits to this approach.
+
+* Treat the scale groups as immutable infrastructure
+* Roll back strategy is as simple as scaling the old scale group back to capacity
+* Managing hosts can be done by scale group or by scale group tags
+
+## Downside
+
+* Managing the compute name or version
+* Risk of removing scale groups
+* Process surrounding acceptance of nodes (not specific to this strategy)
+
+## User Story
+As an admin of OpenShift-Ansible,
+I want to upgrade my scale groups
+so that I can receive product updates, CVE remediations, and provide a path forward for customers.
+
+## Acceptance Criteria
+* Verify that a new scale group is created
+* Verify that the old scale group properly drained
+* Verify that the old scale group is removed
+* Verify that the old launch config is removed
+* Verify there is documentation about the process
+
+## References
+
+
+## Workflow
+
+Outline of a typical workflow for an upgrade.
+* An AMI is built with a specific version of openshift
+  This is done through openshift-ansible/playbooks/aws/openshift-cluster/build_ami.yml playbook.
+  Specify the correct repository location and version via the following vars:
+  openshift_pkg_version=
+  openshift_additional_repositories=[]
+* The output of the build_ami should be an AMI that is placed in the AWS account.
+  The following upgrade playbooks will handle the AMI in two ways:
+  * If the openshift_aws_ami is specified then the scale groups will use the specific id.
+  * If multiple AMIs are desired then you can place them in the the openshift_aws_ami_map:
+  ```
+  openshift_aws_ami_map:
+    master: "{{ openshift_aws_ami }}"
+    infra: "{{ openshift_aws_ami }}"
+    compute: "{{ openshift_aws_ami }}"
+  ```
+  * If the openshift_aws_ami is unspecified, the AWS code will search for the latest
+    AMI with the following name and use it:
+  ```
+  openshift_aws_ami_name: openshift-gi
+  ```
+* The next step is to perform the update on the scale node groups.  The node scale groups are
+  inventory driven.  By default there are two groups that are created which are compute 
+  and infra groups. In order for the upgrade to process correctly, you will specify 
+  the current_version and new_version variables.  These are used to generate the 
+  launch config and the scale group name.
+  ```
+  openshift_aws_current_version: ''
+  openshift_aws_new_version: ''
+  ```
+
+  Theorethically, what this upgrade will look like is the following:
+  ```
+  launch_config=clusterid-type-version
```

Comment:
is "launch_config" an inventory variable?  If so, shouldn't it have an openshift_ prefix on it? if not, where does it go?

Reply:
This is where the version variables tie into the launch config and the scale group.  They create the `clusterid`-`type`-`version`.